pkg/plugins/policies/meshtrafficpermission/graph/reachable_graph.go

```diff
@@ -0,0 +1,72 @@
+package graph
+
+import (
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	ms_api "github.com/kumahq/kuma/pkg/core/resources/apis/meshservice/api/v1alpha1"
+	core_rules "github.com/kumahq/kuma/pkg/plugins/policies/core/rules"
+	mtp_api "github.com/kumahq/kuma/pkg/plugins/policies/meshtrafficpermission/api/v1alpha1"
+	"github.com/kumahq/kuma/pkg/plugins/policies/meshtrafficpermission/graph/backends"
+	graph_backends "github.com/kumahq/kuma/pkg/plugins/policies/meshtrafficpermission/graph/backends"
+	graph_services "github.com/kumahq/kuma/pkg/plugins/policies/meshtrafficpermission/graph/services"
+	"github.com/kumahq/kuma/pkg/xds/context"
+)
+
+var SupportedTags = map[string]struct{}{
+	mesh_proto.KubeNamespaceTag: {},
+	mesh_proto.KubeServiceTag:   {},
+	mesh_proto.KubePortTag:      {},
+}
+
+type Graph struct {
+	rules        map[string]core_rules.Rules
+	backendRules map[backends.BackendKey]core_rules.Rules
+}
+
+func NewGraph(rules map[string]core_rules.Rules, backendRules map[backends.BackendKey]core_rules.Rules) *Graph {
+	return &Graph{
+		rules:        rules,
+		backendRules: backendRules,
+	}
+}
+
+func (r *Graph) CanReach(fromTags map[string]string, toTags map[string]string) bool {
+	if _, crossMeshTagExist := toTags[mesh_proto.MeshTag]; crossMeshTagExist {
```

Comment:
Just a question, can `fromTags` also have key `mesh_proto.MeshTag` and it equals to that in `toTags`?

Reply:
`mesh_proto.MeshTag` is used only for cross mesh traffic and we don't really match it(cross mesh is only on the gateway)